include/deal.II/lac/read_write_vector.templates.h

```diff
@@ -124,6 +124,19 @@ namespace LinearAlgebra
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+  template <typename Number>
+  template <typename Functor>
+  void
+  ReadWriteVector<Number>::apply(const Functor &func)
+  {
+    FunctorTemplate<Functor> functor(*this, func);
+    internal::parallel_for(functor, n_elements(), thread_loop_partitioner);
```

Comment:
OK. I suspect the function to be primarily used for internal work.

Alternatively, you could of course also make the argument of type `std::function<void (number &)>`, in which case it is no longer a template of the functor type and so does not need to be instantiated from user code.